app/assets/javascripts/graph.js

```diff
@@ -0,0 +1,53 @@
+window.graphs = {};
+
+var Graph = (function () {
```

Comment:
I'm a bit late to the party in bringing this up, but... these are *charts* rather than graphs. I know that sounds like nitpicking, but we regularly use the term "graph" to refer to the data structure, and "chart" to mean front-end visuals, and ETModel is consistent in the use of "chart". I think it risks confusion to mix these two terms up.